[PE-D] Long entry shows as grey box

![Screenshot 2022-04-01 at 2.44.51 PM.png](https://raw.githubusercontent.com/jaegarpoon/ped/main/files/6a814409-089d-4c91-a300-3a83af5ced3e.png)

Command entered:
`add n/naaame p/6312896312783612978231912376312876312712368912360326023761230612307812302360231767238061237 e/asg@das a/lol pf/east, 4-room, $99999999999999999999999932189732189712308923101239, $1`

Unable to view user details


<!--session: 1648793093492-a72546db-321d-4806-9933-f3f52d6f032e-->
<!--Version: Web v3.4.2-->

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: jaegarpoon/ped#7

The following exception is thrown when the command is entered:
`Caused by: java.lang.NumberFormatException: For input string: "99999999999999999999999932189732189712308923101239"
`

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -99,7 +99,7 @@ Some example commands you can try:
   - REGION: One of [`North`, `South`, `East`, `West`, `Central`] (Non case-sensitive).
   - ADDRESS: Any non-empty string that does not contain `,`. e.g. `Pasir Ris Drive 1 Block 123`
   - SIZE: One of [`1-room`,`2-room`, `3-room`, `4-room`, `5-room`] (Non case-sensitive).
-  - PRICE: `$` followed by a positive integer. e.g. `$150000`
+  - PRICE: `$` followed by a positive integer with at most 9 digits. e.g. `$150000` is acceptable but `$1234567890` is not acceptable.
 
 </div>
 

--- a/src/main/java/seedu/address/model/property/Price.java
+++ b/src/main/java/seedu/address/model/property/Price.java
@@ -10,7 +10,7 @@ import static seedu.address.commons.util.AppUtil.checkArgument;
 public class Price implements Comparable<Price> {
 
     public static final String MESSAGE_CONSTRAINTS = "Price must have the format $#####";
-    public static final String VALIDATION_REGEX = "^\\$\\d+";
+    public static final String VALIDATION_REGEX = "^\\$\\d{1,9}";
 
     private static final double THOUSAND = 1E3;
     private static final double MILLION = 1E6;

--- a/src/test/java/seedu/address/model/property/PriceTest.java
+++ b/src/test/java/seedu/address/model/property/PriceTest.java
@@ -60,7 +60,6 @@ public class PriceTest {
         assertTrue(Price.isValidPrice("$1000000")); // 10^6
         assertTrue(Price.isValidPrice("$10000000")); // 10^7
         assertTrue(Price.isValidPrice("$100000000")); // 10^8
-        assertTrue(Price.isValidPrice("$1000000000")); // 10^9
     }
 
     @Test

